src/ja/nicomanga/src/eu/kanade/tachiyomi/extension/ja/nicomanga/Nicomanga.kt

```diff
@@ -0,0 +1,165 @@
+package eu.kanade.tachiyomi.extension.ja.nicomanga
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import java.net.URL
+
+class Nicomanga : HttpSource() {
+    companion object {
+        val thumbnailURLRegex: Regex = "background-image:[^;]url\\s*\\(\\s*'([^?']+)".toRegex()
+
+        val statusRegex: Regex = "(?<=-)[^.]+".toRegex()
+
+        val urlRegex: Regex = "(?<=manga-)[^/]+(?=\\.html\$)".toRegex()
+
+        val floatRegex: Regex = "\\d+(?:\\.\\d+)?".toRegex()
+
+        val chapterIdRegex: Regex = "(?<=imgsListchap\\()\\d+".toRegex()
+    }
```

Comment:
Should make these fields private. Could also move them out of the `companion object` since the source class is only created once.

Also you're using lookbehind assertions `(?<=)` too much. Replace them with simpler expressions with `()` to capture what you need.